plugins/auth-backend-module-oidc-proxy-provider/src/resolvers.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createSignInResolverFactory,
+  SignInInfo,
+} from '@backstage/plugin-auth-node';
+import {
+  stringifyEntityRef,
+  DEFAULT_NAMESPACE,
+} from '@backstage/catalog-model';
+import { OidcProxyResult } from './types';
+
+/**
+ * Available sign-in resolvers for the auth provider to assoicate oidc id tokens
+ * with a backstage User entity.
+ *
+ * @public
+ */
+export namespace oidcProxySignInResolvers {
+  /**
+   * singInWithoutCatalogUser signs the user in without requiring a pre-existing
+   * User entity in the catalog.  Refer to
+   * https://backstage.io/docs/auth/identity-resolver/#sign-in-without-users-in-the-catalog
+   */
+  export const signInWithoutCatalogUser = createSignInResolverFactory({
```

Comment:
The name is a bit generic - may make sense to name it `emailLocalPartAsSubjectWithoutCatalogUser` or `trustEmailLocalPartAsSubject` or something like that? To differentiate e.g. something that uses the `sub` claim instead. Naming is hard :)

Reply:
Will rename to `emailLocalPartAsSubjectWithoutCatalogUser`, I like it.  It is what is says it is.

Reply:
Fixed in 84309f706907 `oidc-proxy-provider: clarify sign in provider config`